addons/sale_loyalty/models/sale_order.py

```diff
@@ -410,7 +411,7 @@ def _discountable_specific(self, reward):
             line_reward = lines.reward_id
             discounted_lines = order_lines
             if line_reward.discount_applicability == 'cheapest':
-                cheapest_line = cheapest_line or self._cheapest_line()
+                cheapest_line = cheapest_line or self._cheapest_line(line_reward)
```

Comment:
I feel like this line could use a comment why in discount_specific we check cheapest line.